azext_iot/iothub/providers/state.py

```diff
@@ -51,7 +56,15 @@ def __init__(
             login=login,
             auth_type_dataplane=auth_type_dataplane
         )
+
+        if not login:
+            self.include_control_plane = True
+            self.rg = self.target["resourcegroup"]
```

Comment:
does this actually get populated? the connection string should not have a resource group and the discovery builds on that, so this should be same as self.rg = None

put prints and double check

Reply:
Not sure I understand. **self.rg = self.target["resourcegroup"]** is only running if the connection string isn't provided. If the user only gives the hub name and not the resource group, this is getting the name of the resource group.